api/WebApi.ts

```diff
@@ -42,7 +42,7 @@ import os = require('os');
 import url = require('url');
 import path = require('path');
 
-const isBrowser: boolean = typeof window !== 'undefined';
+const isBrowser: boolean = typeof window !== 'undefined' && Boolean(window.navigator);
```

Comment:
`isBrowser` is used in other places that don't rely on `window.navigator`. Would it be better to localize the boolean condition just to the part trying to find the user agent string?

Reply:
I mean, wouldn't it be better for `isBrowser` to not think our application is... a browser...?

Reply:
The only other place it's used is to throw an error on `_readTaskLibSecrets`, which looks like it'd blow up the instance of the API anyways if it managed to get in there while thinking it's a browser.

Reply:
Is there a more robust way we could see if it were an electron main process? For example we could look at [process](https://www.electronjs.org/docs/api/process) and say "ok, maybe there's a window, but `process && process.type && process.type !== 'browser'` so actualy _no_ we're not in a browser."

Reply:
Potentially, yeah.

Reply:
@Mr-Wallet Done, I still kept the `isNode` variable separate though cause I think it makes sense for readers to know what the goal is of that piece of the conditional.